File: java/poker/src/main/java/com/duchen/poker/model/type/ZeroTypeChecker.java
package com.duchen.poker.model.type;

import java.util.Arrays;
import java.util.List;

public class ZeroTypeChecker implements TypeChecker {

    @Override
    public CardGroupType getType() {
        return CardGroupType.ZERO;
    }

    @Override
    public List<Integer> getMyPossibleCardCount() {
        return Arrays.asList(0);
    }

    @Override
    public int getMaxCardIfIsMyType(List<Integer> cards) {
        return 0;
    }

}

File: java/poker/src/main/kotlin/poker/model/type/ZeroTypeChecker.kt
package poker.model.type

import java.util.Arrays

class ZeroTypeChecker : TypeChecker {

    override val type: CardGroupType
        get() = CardGroupType.ZERO

    override val myPossibleCardCount: List<Int>
        get() = Arrays.asList(0)

    override fun getMaxCardIfIsMyType(cards: MutableList<Int>): Int {
        return 0
    }

}
